idx/memory/meta_tags.go

```diff
@@ -0,0 +1,236 @@
+package memory
+
+import (
+	"fmt"
+	"hash"
+	"hash/fnv"
+	"sort"
+	"strings"
+)
+
+// the collision avoidance window defines how many times we try to find a higher
+// slot that's free if two record hashes collide
+const collisionAvoidanceWindow = 3
+
+// the function we use to get the hash for hashing the meta records
+// it can be replaced for mocking in tests
+var queryHash func() hash.Hash32
+
+func init() {
+	queryHash = fnv.New32a
+}
+
+type metaTagRecord struct {
+	metaTags []kv
```

Comment:
In a future PR the `.cost` field will be removed from `kv`, to make it look like this: https://github.com/grafana/metrictank/pull/960/files#diff-ee1d033820a25c488854de4501fc93a4L50
You're right though that at this current state of this branch, the `.cost` is just a waste of memory inside this structure. I can't remove it from the `kv` type because it's still needed in the normal query execution, but I could temporarily create a new type that could be used here instead of `kv`.

Reply:
ok whatever you think is best/easiest